src/ssh/azext_ssh/ssh_utils.py

```diff
@@ -106,6 +124,8 @@ def _read_ssh_logs(ssh_sub, op_info, delete_cert, delete_keys):
            not next_line.startswith("Authenticated "):
             sys.stderr.write(next_line)
             _check_for_known_errors(next_line, delete_cert, log_list)
+            if not service_config_delay_error:
+                service_config_delay_error = check_for_service_config_delay_error(next_line)
 
         if "debug1: Entering interactive session." in next_line:
             connection_established = True
```

Comment:
Maybe before doing cleanup on line 137 we can make sure that service config delay didn't happen? I don't think that would ever happen, but when we need to rely on timers to synchronize operations I always get a little worried about race conditions.

Reply:
I'll think some more about this suggestion.

Reply:
Done - I moved the cleanup on line 137 outside of the while loop though so that the `service_config_delay_error` has been updated based on all the available logs

Reply:
If you move that cleanup outside of the for loop, then the credentials will only be cleaned up once the Process exits. The goal of that cleanup is to clean up the credentials after 2 minutes if the process hasn't exited yet, because that likely means that the connection was established, but we failed to identify that (basically is just to make sure that if the log messages change in future releases of OpenSSH we still delete the credentials once they are no longer needed).
I think just having the "not service_config_delay_error" check would be enough

Reply:
ok moved back inside the while loop